community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreLabelCursor.java

```diff
@@ -61,28 +63,24 @@ public boolean next()
     }
 
     @Override
-    public boolean seek( int labelId )
+    public LabelItem get()
     {
-        while ( next() )
-        {
-            if ( currentLabel == labelId )
-            {
-                return true;
-            }
-        }
-
-        return false;
+        return labelItem;
     }
 
     @Override
-    public int getLabel()
+    public void close()
     {
-        return currentLabel;
+        instanceCache.accept( this );
     }
 
-    @Override
-    public void close()
+    private class StoreLabelItem
```

Comment:
Why do we need inner class here? Can just `StoreLabelCursor` implement `LabelItem`?